docs/tasks/debug-application-cluster/local-debugging.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Developing and debugging services locally
+---
+
+{% capture overview %}
+
+Kubernetes applications usually consist of multiple, separate services, each running in its own container. Developing and debugging these services on a remote Kubernetes cluster can be cumbersome, requiring you to [get a shell on a running container](https://kubernetes.io/docs/tasks/debug-application-cluster/get-shell-running-container/) and then running your tools inside the remote shell.
+
+This document describes using `telepresence` to develop and debug a service locally, while proxying the service to a remote Kubernetes cluster. Using `telepresence` allows you to use custom tools, such as a debugger and IDE, for a local service and provides the service full access to ConfigMap, secrets, and the services running on the remote cluster.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* Kubernetes cluster is installed
+* `kubectl` is configured to communicate with the cluster
+* [Telepresence](https://www.telepresence.io/reference/install) is installed
+
+{% endcapture %}
+
+{% capture steps %}
+
+## Getting a shell on a remote cluster
+
+If you start `telepresence` with no arguments, you'll get a special shell. Unlike a remote shell, the telepresence shell runs locally, giving you full access to your filesystem.
+
+There are a variety of ways that the local shell can be used. For example, you could write a shell script on your laptop, and run it directly from the shell in real-time. You can do this on a remote shell as well, of course, but you may not be able to use your preferred code editor, and the script is deleted when you terminate the container.
```

Comment:
The `telepresence` shell can be used in a variety of ways. For example, write a shell script on your laptop, and run it directly from the shell in real time. You can do this on a remote shell as well, but you might not be able to use your preferred code editor, and the script is deleted when the container is terminated.